lib/data/entity.js

```diff
@@ -38,9 +38,26 @@ const odataToColumnMap = new Map([
 const validateEntity = (entity) => {
   const { label, id, update, baseVersion } = entity.system;
 
-  if (!label || label.trim() === '')
+  // Trying to describe this logic.
+  // if coming from a submission, create and/or update will hopefully be defined.
+  // if coming from json, create and update will not be available.
+  // in most cases, we need label to exist and not be blank...
+  // - creating an entity with a label
+  // - updating an entity's label
+  // but in one case, if its an update and label is not even specified,
+  // that is ok -- the label just wont be updated on the entity.
+  // TODO: do something different with validateEntity because it has too
+  // many special, subtle cases.
+  if (!(update === '1' || update === 'true') && (!label || label.trim() === ''))
```

Comment:
see totally new logic - instead of calling this `validateEntity` function that tries to do it all, while also manipulating the structure, the each field is pulled out and validated on its own to build up the new entity def.